src/storable.rs

```diff
@@ -421,6 +421,44 @@ where
     };
 }
 
+impl<T: Storable> Storable for Option<T> {
+    fn to_bytes(&self) -> Cow<[u8]> {
+        match self {
+            Some(t) => {
+                let mut bytes = t.to_bytes().into_owned();
+                bytes.push(1);
+                Cow::Owned(bytes)
+            }
+            None => Cow::Borrowed(&[0]),
+        }
+    }
+
+    fn from_bytes(bytes: Cow<[u8]>) -> Self {
+        match bytes.split_last() {
+            Some((&last, bytes)) => match last == 0 {
+                true => None,
+                false => Some(T::from_bytes(Cow::Borrowed(bytes))),
+            },
+            None => {
+                panic!("Invalid Option encoding");
+            }
```

Comment:
Let's be stricter in the decoding to help users discover their errors as soon as possible. For example, we shouldn't accept `[1, 0]` as `None` and shouldn't accept byte sequences that don't end by `{0, 1}` (e.g., `[0, 3]`).
```suggestion
            Some((last, rest)) => match last {
                0 => {
                    assert!(rest.is_empty(), "Invalid Option encoding: unexpected prefix before the None marker: {rest:?}");
                    None
                }
                1 => Some(T::from_bytes(Cow::Borrowed(rest))),
                _ => panic!("Invalid Option encoding: unexpected variant marker {last}"),
            },
            None => panic!("Invalid Option encoding: expected at least one byte"),

```